File: main/java/archbots/nullhide/leaves/Logout.java
package archbots.nullhide.leaves;

import com.runemate.game.api.hybrid.Environment;
import com.runemate.game.api.hybrid.RuneScape;
import archapi.treebot.LeafTask;

public class Logout extends LeafTask {
    @Override
    public void execute() {
        RuneScape.logout();
        Environment.getBot().stop("Out of supplies");
    }
}

File: arch/api/tasks/general/Logout.kt
package arch.api.tasks.general

import arch.api.treebot.LeafTask
import com.runemate.game.api.hybrid.Environment
import com.runemate.game.api.hybrid.RuneScape
import com.runemate.game.api.script.Execution

class Logout: LeafTask(){
    override fun execute() {
        RuneScape.logout()
        Execution.delayUntil({!RuneScape.isLoggedIn()})
        Environment.getBot().stop("Finished tasks")
    }
}